tests/test_paperqa.py

```diff
@@ -1239,6 +1240,69 @@ def test_dois_resolve_to_correct_journals(doi_journals):
     assert details.journal == doi_journals["journal"]
 
 
+@pytest.mark.asyncio
+async def test_docdetails_merge_with_non_list_fields():
+    # Test merging a DocDetail that was republished
+    # (so it has a new publication date but represents the same document)
+    # Create two DocDetails instances
+    doc1 = DocDetails(
+        citation="Citation 1",
+        publication_date=datetime(2023, 1, 1),
+        docname="Document 1",
+        dockey="key1",
+        other={"bibtex_source": "source1", "client_source": ["client1"]},
+    )
+
+    doc2 = DocDetails(
+        citation="Citation 1",
+        publication_date=datetime(2024, 1, 1),
+        docname="Document 1",
+        dockey="key2",
+        other={"bibtex_source": ["source2"], "client_source": "client2"},
+    )
```

Comment:
```suggestion
    # Test merging a DocDetail that was republished
    doc1 = DocDetails(
        citation="Citation 1",
        publication_date=datetime(2023, 1, 1),
        docname="Document 1",
        dockey="key1",
        other={"bibtex_source": "source1", "client_source": ["client1"]},
    )
    republished_next_quarter = doc1.publication_date + timedelta(weeks=14)
    doc2 = DocDetails(
        citation=doc1.citation,
        publication_date=republished_next_quarter,
        docname=doc1.docname,
        dockey="key2",
        other={"bibtex_source": ["source2"], "client_source": "client2"},
    )
```

Here is how to do the code in a more self-documenting manner:
- Propagating the attributes that don't matter
- Using a descriptive self-documenting name for `publication_date`

Reply:
Great idea! Thanks for letting me know